docs/content/schemas.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Type Checking
+kind: misc
+weight: 2
+---
+
+## Using schemas to enhance the Rego type checker
+
+You can provide an input schema to `opa eval` to improve static type checking and get more precise error reports as you develop Rego code.
+The `-s` flag can be used to upload a single schema for the input document in JSON Schema format.
+
+```
+-s, --schema string set schema file path
+```
+
+Example:
+```
+opa eval data.envoy.authz.allow -i example/envoy/input.json -d example/envoy/policy.rego -s example/envoy/input-schema.json
+```
+Samples provided at: https://github.com/aavarghese/opa-schema-examples/tree/main/envoy
+
+
+{{< figure src="OPATypeChecking_PR3053.gif" width="130" caption="Demo" >}}
```

Comment:
This will become part of the docs page, I think. Can we drop the gif animation, please? I think the _Usage Scenario_ outlined below is a nice, self-contained introduction. The page could then end with a _reference_ section pointing towards
- the opa-schema-examples repo,
- and maybe some official jsonschema docs page?

Reply:
I do think that should be publicised widely, and the gif example could help for quickly grasping what this is about. But I think I'd prefer it not in the docs... What do you think? 😃

Reply:
Done (commit https://github.com/open-policy-agent/opa/pull/3060/commits/0e0d9152740e6f6d9035a6bb6250ee0bf24d493e)

Perhaps the gif could be useful for release notes or we could include it in a blog...

Reply:
I think we should definitely do a blog post here! 🚀